### PR TITLE
extension with LoV (#731)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- RN: Support inner_properties for Element: properties that are only internal and used to handle communication between the backend and the frontend (no user visibility)
- refresh typed variable (_TaipyBase)
- LoV adapter: allow extra properties in LoV element

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Review by FabienLelaquais — APPROVED** [review]
I get it now.
Well done!

**Comment by FredLL-Avaiga** [comment]
@dinhlongviolin1 Any idea why gui.pyi file is causing pb with mypy ?

File: src/taipy/gui/config.py [hunk 1]
```diff
@@ -159,9 +159,7 @@ def _get_config(self, name: ConfigParameter, default_value: t.Any) -> t.Any:  #
                 try:
                     return type(default_value)(self.config[name])
                 except Exception as e:
-                    warnings_warn(
-                        f'app_config "{name}" value "{self.config[name]}" is not of type {type(default_value)}\n{e}'
-                    )
+                    _warn(f'app_config "{name}" value "{self.config[name]}" is not of type {type(default_value)}\n{e}')
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/config.py]: no :\n{e} ?

File: src/taipy/gui/utils/date.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def _date_to_ISO(date_val: t.Union[datetime, date, time]) -> str:
         except Exception as e:
             # astimezone() fails on Windows for pre-epoch times
             # See https://bugs.python.org/issue36759
-            warnings_warn(f"There is some problems with date parsing to ISO!\n{e}")
+            _warn(f"There is some problems with date parsing to ISO!\n{e}")
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/utils/date.py]: ! => :

File: src/taipy/gui/config.py [hunk 3]
```diff
@@ -227,7 +225,7 @@ def _build_config(self, root_dir, env_filename, kwargs):  # pragma: no cover
                     else:
                         config[key] = value if config[key] is None else type(config[key])(value)  # type: ignore
                 except Exception as e:
-                    warnings_warn(
+                    _warn(
                         f"Invalid keyword arguments value in Gui.run {key} - {value}. Unable to parse value to the correct type.\n{e}"
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/config.py]: :

File: src/taipy/gui/config.py [hunk 4]
```diff
@@ -238,8 +236,8 @@ def _build_config(self, root_dir, env_filename, kwargs):  # pragma: no cover
                     try:
                         config[key] = value if config[key] is None else type(config[key])(value)  # type: ignore
                     except Exception as e:
-                        warnings_warn(
-                            f"Invalid env value in Gui.run: {key} - {value}. Unable to parse value to the correct type.\n{e}"
+                        _warn(
+                            f"Invalid env value in Gui.run(): {key} - {value}. Unable to parse value to the correct type.\n{e}"
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/config.py]: :

File: src/taipy/gui/config.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/config.py]: How and why did I miss that?!?
Thx!

File: src/taipy/gui/utils/date.py (hunk 2 above)
Comment (FabienLelaquais) [reply src/taipy/gui/utils/date.py]: 👌

File: src/taipy/gui/config.py (hunk 3 above)
Comment (FabienLelaquais) [reply src/taipy/gui/config.py]: 👌

**Pull request merged**